src/Magento/FunctionalTestingFramework/StaticCheck/DeprecatedEntityUsageCheck.php

```diff
@@ -0,0 +1,605 @@
+     *              'get',
+     *              'create',
+     *              'update',
+     *          ],
+     *      ],
+     *      ...
+     *  ]
+     *
+     *  or
+     *
+     *  [
+     *      $dataName1 => [
+     *          $metaDataName1 => 'all',
+     *      ],
+     *      $dataName2 => [
+     *          $metaDataName2 => 'all',
+     *      ],
+     *      $dataName5 => [
+     *          $metaDataName5 => 'all',
+     *      ],
+     *      ...
+     *  ]
+     *
+     * @param array $references
+     * @return array
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    private function findViolatingMetadataReferences($references)
+    {
+        $allObjects = OperationDefinitionObjectHandler::getInstance()->getAllObjects();
+
+        // Find Violations
+        $violatingReferences = [];
+        foreach ($references as $dataName => $metadataArray) {
+            foreach ($metadataArray as $metadata => $types) {
+                $operations = [];
+                $strict = true;
+                if (is_array($types)) {
+                    $operations = $types;
+                } elseif ($types === 'all') {
+                    $operations = $this->dataOperations;
+                    $strict = false;
+                } else {
+                    $operations = [$types];
+                }
+
+                $deprecated = null;
+                $exists = false;
+                foreach ($operations as $operation) {
+                    if (array_key_exists($operation . $metadata, $allObjects)) {
+                        $exists = true;
+                        /** @var OperationDefinitionObject $entity */
+                        $entity = $allObjects[$operation . $metadata];
+                        // When not strictly checking, it's not deprecated as long as we found one that's not deprecated
+                        if (!$strict && empty($entity->getDeprecated())) {
+                            $deprecated = false;
+                            break;
+                        }
+                        // When strictly checking, it's deprecated as long as we found one that's deprecated
+                        if ($strict && !empty($entity->getDeprecated())) {
+                            $deprecated = true;
+                            break;
+                        }
+                    }
+                }
+                if ($exists && !$strict && $deprecated !== false) {
+                    $deprecated = true;
+                }
+                if ($strict && $deprecated !== true) {
+                    $deprecated = false;
+                }
+
+                if ($deprecated) {
+                    $violatingReferences["\"{$dataName}\" references deprecated"][] = [
+                        'name' => $metadata,
+                        // TODO add filename in OperationDefinitionObject
+                        'file' => 'metadata xml file',
```

Comment:
Just double checking on the TODO, is this intended to be fixed in a later release?

Reply:
Metadata does not have filename information. This is more for future work.